src/analysis/calculus/conformal.lean

```diff
@@ -0,0 +1,184 @@
+-/
+import analysis.normed_space.conformal_linear_map
+
+/-!
+# Conformal Maps
+
+A continuous linear map between real normed spaces `X` and `Y` is `conformal_at` some point `x`
+if it is real differentiable at that point and its differential `is_conformal_linear_map`.
+
+## Main definitions
+
+* `conformal_at`: the main definition of conformal maps
+* `conformal`: maps that are conformal at every point
+* `conformal_factor_at`: the conformal factor of a conformal map at some point
+
+## Main results
+* The conformality of the composition of two conformal maps, the identity map
+  and multiplications by nonzero constants
+* `conformal_at_iff`: an equivalent definition of the conformality
+* `conformal_at.preserves_angle`: if a map is conformal at `x`, then its differential
+                                  preserves all angles at `x`
+
+## Tags
+
+conformal
+
+## Warning
+
+The definition of conformality in this file does NOT require the maps to be orientation-preserving.
+Maps such as the complex conjugate is considered as a conformal map.
+-/
+
+noncomputable theory
+
+variables {E F : Type*} [inner_product_space ℝ E] [inner_product_space ℝ F]
+  {X Y Z : Type*} [normed_group X] [normed_group Y] [normed_group Z]
+  [normed_space ℝ X] [normed_space ℝ Y] [normed_space ℝ Z]
+
+section loc_conformality
+
+open linear_isometry continuous_linear_map
+open_locale real_inner_product_space
+
+/-- A map `f` is said to be conformal if it has a conformal differential `f'`. -/
+def conformal_at (f : X → Y) (x : X) :=
+∃ (f' : X →L[ℝ] Y), has_fderiv_at f f' x ∧ is_conformal_map f'
+
+lemma conformal_at_id (x : X) : conformal_at id x :=
+⟨id ℝ X, has_fderiv_at_id _, is_conformal_map_id⟩
+
+lemma conformal_at_const_smul {c : ℝ} (h : c ≠ 0) (x : X) :
+  conformal_at (λ (x': X), c • x') x :=
+⟨c • continuous_linear_map.id ℝ X,
+  has_fderiv_at.const_smul (has_fderiv_at_id x) c, is_conformal_map_const_smul h⟩
+
+/-- A real differentiable map `f` is conformal at point `x` if and only if
+    its differential `f'` at that point scales any inner product by a positive scalar. -/
+lemma conformal_at_iff {f : E → F} {x : E} {f' : E →L[ℝ] F}
+  (h : has_fderiv_at f f' x) : conformal_at f x ↔ ∃ (c : ℝ) (hc : 0 < c),
+  ∀ (u v : E), ⟪f' u, f' v⟫ = (c : ℝ) * ⟪u, v⟫ :=
+begin
+  split,
+  { rintros ⟨f₁, hf, c₁, hc₁, li, hf₁⟩,
+    exact (hf.unique h) ▸ (is_conformal_map_iff f₁).mp ⟨c₁, hc₁, li, hf₁⟩, },
+  { rintros ⟨c, hc, huv⟩,
+    exact ⟨f', h, (is_conformal_map_iff f').mpr ⟨c, hc, huv⟩⟩, },
+end
+
+namespace conformal_at
+
+lemma differentiable_at {f : X → Y} {x : X} (h : conformal_at f x) :
+  differentiable_at ℝ f x :=
+let ⟨_, h₁, _⟩ := h in h₁.differentiable_at
+
+lemma congr {f g : X → Y} {x : X} {u : set X} (hx : x ∈ u) (hu : is_open u)
+  (hf : conformal_at f x) (h : ∀ (x : X), x ∈ u → g x = f x) :
+  conformal_at g x :=
+let ⟨f', hfderiv, hf'⟩ := hf in
+  ⟨f', hfderiv.congr_of_eventually_eq (filter.eventually_eq_of_mem (hu.mem_nhds hx) h), hf'⟩
```

Comment:
```suggestion
  ⟨f', hfderiv.congr_of_eventually_eq ((hu.eventually_mem hx).mono h), hf'⟩
```

Reply:
Thank you for correcting these trivial things. I should've done them myself.